cradle-cassandra/src/main/java/com/exactpro/cradle/cassandra/dao/messages/MessageBatchOperator.java

```diff
@@ -70,10 +70,16 @@ DetailedMessageBatchEntity get(UUID instanceId, String streamName, String direct
 	@Select
 	PagingIterable<DetailedMessageBatchEntity> getAll(Function<BoundStatementBuilder, BoundStatementBuilder> attributes);
 	
+	@Query("SELECT * FROM ${qualifiedTableId} WHERE "
+			+INSTANCE_ID+"=:instanceId AND "+STREAM_NAME+"=:streamName AND "+DIRECTION+"=:direction LIMIT 1")
+	DetailedMessageBatchEntity getFirstIndex(UUID instanceId, String streamName, String direction,
```

Comment:
For this particular query you don't need to get the whole message batch, only message_index value as Long.

Reply:
The method with long or Long return type will fail if the result set is empty.

Reply:
But I can use Row type without mapping